sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/InMemoryRelation.scala

```diff
@@ -200,11 +202,14 @@ case class InMemoryRelation(
   }
 
   override def computeStats(): Statistics = {
-    if (cacheBuilder.sizeInBytesStats.value == 0L) {
+    if (!cacheBuilder.isCachedColumnBuffersLoaded) {
       // Underlying columnar RDD hasn't been materialized, use the stats from the plan to cache.
       statsOfPlanToCache
     } else {
-      statsOfPlanToCache.copy(sizeInBytes = cacheBuilder.sizeInBytesStats.value.longValue)
+      statsOfPlanToCache.copy(
+        sizeInBytes = cacheBuilder.sizeInBytesStats.value.longValue,
+        rowCount = Some(cacheBuilder.rowCountStats.value.longValue)
```

Comment:
Is `rowCount` required additionally?

Reply:
Yea, we need it because this change passes `rowCount` into upper nodes;
```
scala> sql("CREATE VIEW v AS SELECT 1 c")
scala> sql("CACHE TABLE v")
scala> spark.table("v").explain(true)
...
== Optimized Logical Plan ==
InMemoryRelation [c#28], StorageLevel(disk, memory, deserialized, 1 replicas)
   +- *(1) Project [1 AS c#1]
      +- Scan OneRowRelation[]
...

> w/o this change
scala> val stats = spark.table("v").queryExecution.optimizedPlan.stats
.... Statistics(sizeInBytes=4.0 B)

> w/ this change
scala> val stats = spark.table("v").queryExecution.optimizedPlan.stats
.... Statistics(sizeInBytes=4.0 B, rowCount=1)
                                   ^^^^^^^^^^^
```